pkg/grpc/requestinfo/headers_multimap.go

```diff
@@ -17,3 +18,25 @@ type AsHeadersMultiMap http.Header
 func (h AsHeadersMultiMap) Get(key string) []string {
 	return http.Header(h).Values(key)
 }
+
+// Set implements HeadersMultiMap interface.
```

Comment:
Could you please add a type check here that asserts that the `HeadersMultiMap` interface is fulfilled by `AsHeadersMultiMap`?